src/core/include/mp-units/framework/representation_concepts.h

```diff
@@ -309,16 +310,46 @@ concept VectorRepresentation = (!is_quantity<T>) && Vector<T> && requires(const
 
 MP_UNITS_EXPORT template<typename T>
 concept Representation = detail::ScalarRepresentation<T> || detail::ComplexRepresentation<T> ||
-                         detail::VectorRepresentation<T>;  // || detail::TensorRepresentation<T>;
+                         detail::VectorRepresentation<T>;  // || detail::TensorRepresentation<T>; */
 
 namespace detail {
 
+#ifdef MP_UNITS_APPLE_CLANG_HACKS
+template<typename T>
+constexpr bool is_weakly_regular = std::copyable<T> && std::equality_comparable<T>;
+
+template<typename T>
+constexpr bool is_scalar = !disable_scalar<T> && is_weakly_regular<T>;
+
+template<typename T>
+constexpr bool is_complex = !disable_complex<T> && is_weakly_regular<T> && is_scalar<value_type_t<T>> &&
+                            std::constructible_from<T, value_type_t<T>, value_type_t<T>>;
+
+template<typename T>
+concept ComplexFunctionsAvailable = requires(T a) {
+  ::mp_units::real(a);
+  ::mp_units::imag(a);
+  ::mp_units::modulus(a);
+};
+
+template<typename T>
+constexpr bool is_vector = !disable_vector<T> && is_weakly_regular<T> && is_scalar<value_type_t<T>>;
```

Comment:
Can we use concepts instead? If we can't use the same names (e.g., `Vector`), then maybe `VectorHack` or something similar can be used. I just don't want to leave the impression that this is a customization point that may be partially specialized somewhere.

Reply:
Sure, concepts are preferred, but if they will not work type traits are also fine.